### PR TITLE
GH-8720: Check MQTT topics if not empty strings

### LINKED ISSUE #8720: The MQTT protocol doesn't allow an empty topic value nor for publisihng neither for subscribing
**In what version(s) of Spring Integration are you seeing this issue?**

For example:

6.0.7

**Describe the bug**

**To Reproduce**

AbstractMqttMessageDrivenChannelAdapter#initTopics
The initTopics method of the AbstractMqttMessageDrivenChannelAdapter class does not check if the topic is an empty string

if empty string

MqttPahoMessageDrivenChannelAdapter#subscribe
The subscription part in the subscribe method in the MqttPahoMessageDrivenChannelAdapter class will report an error
`IMqttToken subscribeToken = this.client.subscribe(topics, requestedQos, listeners);`

**Expected behavior**

A verification mechanism should be added

**Sample**

IMqttToken subscribeToken = this.client.subscribe(topics, requestedQos, listeners);    
Will report an error

### PR DESCRIPTION
**Pull request by artembilan**
Fixes https://github.com/spring-projects/spring-integration/issues/8720

Validate MQTT topics for empty strings in the channel adapters configuration

**Cherry-pick to `6.1.x` & `6.0.x`**

### CODE REVIEW

**Comment by artembilan** [comment]
I'll back-port to `5.5.x` separately: too many breaking changes since then.
Also let me know if this does not cherry-pick properly, so I'll come up with other fix for those versions.

File: spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/AbstractMqttMessageDrivenChannelAdapter.java [hunk 1]
```diff
@@ -277,7 +286,7 @@ public void addTopic(String topic, int qos) {
 	 */
 	@ManagedOperation
 	public void addTopic(String... topic) {
```
Comment (tzolov) [inline spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/AbstractMqttMessageDrivenChannelAdapter.java:288]: For consistency  use the plural naming: `topic` -> `topics`.

File: spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/AbstractMqttMessageDrivenChannelAdapter.java [hunk 2]
```diff
@@ -298,8 +307,7 @@ public void addTopic(String... topic) {
 	 */
 	@ManagedOperation
 	public void addTopics(String[] topic, int[] qos) {
```
Comment (tzolov) [inline spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/AbstractMqttMessageDrivenChannelAdapter.java:309]: Same, for consistency  use the plural naming: `topic` -> `topics`.

File: spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/AbstractMqttMessageHandler.java [hunk 3]
```diff
@@ -358,11 +363,13 @@ public boolean isRunning() {
 	protected void handleMessageInternal(Message<?> message) {
 		Object mqttMessage = this.converter.fromMessage(message, Object.class);
 		String topic = this.topicProcessor.processMessage(message);
-		if (topic == null && this.defaultTopic == null) {
-			throw new IllegalStateException(
-					"No topic could be determined from the message and no default topic defined");
+		if (topic == null) {
```
Comment (tzolov) [inline spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/AbstractMqttMessageHandler.java:366]: Can the `topic` be returned as an empty string?

**Comment by artembilan** [comment]
Back-ported to `5.5.x` as https://github.com/spring-projects/spring-integration/commit/a82a41ece307e1845454fa6d1e6eeebb49aa7c46 after fixing conflicts.

**Pull request merged**